packages/services/cdn-worker/src/is-app-deployment-active.ts

```diff
@@ -0,0 +1,73 @@
+import { type ArtifactStorageReader } from './artifact-storage-reader';
+
+type GetCache = () => Promise<Cache | null>;
+type WaitUntil = (promise: Promise<void>) => void;
+
+export type IsAppDeploymentActive = (
+  targetId: string,
+  appName: string,
+  appVersion: string,
+) => Promise<boolean>;
+
+/** check whether an app deployment is active (optionally using a cache to avoid accessing S3) */
+export function createIsAppDeploymentActive(deps: {
+  artifactStorageReader: ArtifactStorageReader;
+  waitUntil: null | WaitUntil;
+  getCache: null | GetCache;
+}): IsAppDeploymentActive {
+  return async function isAppDeploymentActive(
+    targetId: string,
+    appName: string,
+    appVersion: string,
+  ): Promise<boolean> {
+    const cache = await (deps.getCache ? deps.getCache() : null);
+    const cacheKey = new Request(
+      [
+        'http://key-cache.graphql-hive.com',
+        'v1',
+        targetId,
+        'apps-enabled',
+        appName,
+        appVersion,
+      ].join('/'),
```

Comment:
Maybe an assertion here as well? Not not accidentally cache things under the same key (in case of undefined etc)

Reply:
done!